src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs

```diff
@@ -204,11 +206,31 @@ private void CacheInstanceDiscoveryMetadata(InstanceDiscoveryMetadataEntry metad
 
         private async Task<Uri> BuildAuthorityWithRegionAsync(Uri canonicalAuthority, RequestContext requestContext)
         {
+            string regionToUse = requestContext.ServiceBundle.Config.AuthorityInfo.RegionToUse;
+
             if (s_region.IsNullOrEmpty())
             {
-                s_region = await GetRegionAsync(requestContext).ConfigureAwait(false);
+                try
+                {
+                    s_region = await GetRegionAsync(requestContext).ConfigureAwait(false);
```

Comment:
>s_region [](start = 20, length = 8)

what happens if this is null?

Reply:
Yes, in case the region is not detected an exception is thrown